fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/PermissiveAvailabilitySpec.groovy

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.availability
+
+import com.yahoo.bard.webservice.data.config.names.TableName
+import com.yahoo.bard.webservice.metadata.TestDataSourceMetadataService
+import com.yahoo.bard.webservice.table.Column
+import com.yahoo.bard.webservice.table.resolver.DataSourceConstraint
+import com.yahoo.bard.webservice.util.SimplifiedIntervalList
+
+import org.joda.time.Interval
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import java.util.stream.Collectors
+
+class PermissiveAvailabilitySpec extends Specification  {
+
+    PermissiveAvailability permissiveAvailability
+    TableName tableName
+    Column column1
+    Column column2
+    Interval interval1
+    Interval interval2
+
+    def setup() {
+        tableName = TableName.of('table')
+
+        column1 = new Column('column1')
+        column2 = new Column('column2')
+
+        interval1 = new Interval('2010-01-01/2020-12-31')
+        interval2 = new Interval('2017-01-01/2020-12-31')
```

Comment:
Move the setup things that are not used by more than one test into the single test that uses it.